### DISCUSSION TITLE
Method Not Allowed when i use post

### DISCUSSION THREAD

**Discussion by muhiqsimui** [fastapi/fastapi#6100] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import List
from fastapi import FastAPI
from uuid import UUID,uuid4
from models import User, Gender, Role

app = FastAPI()

db: List[User] = [
    User(
        id=uuid4(),
        first_name="John",
        last_name="Doe",
        gender=Gender.male,
        roles=[Role.user]
         ),
    User(
        id=UUID("3f4f106b-1758-4e1f-a5ac-205ad4a177bf"),
        first_name="Natalia",
        last_name="Landofdawn",
        gender=Gender.female,
        roles=[Role.user,Role.admin]
         )
]

@app.get('/')
async def root():
    return {"hello":"world"}

@app.get('/api/v1/users')
async def fetch_users():
    return db

@app.post("/api/v1/users'")
async def register_user(user: User):
    db.append(user)
    return {"id":user.id}
```

### Description

hello sir i have some problem when i try use method post result {
  "detail": "Method Not Allowed"
}
 here is screenshoot
![image](https://user-images.githubusercontent.com/33776809/211401306-bdf9691c-1799-429f-b745-d1b71ede30d4.png)

this is my code
main.py
`from typing import List
from fastapi import FastAPI
from uuid import UUID,uuid4
from models import User, Gender, Role

app = FastAPI()

db: List[User] = [
    User(
        id=uuid4(),
        first_name="John",
        last_name="Doe",
        gender=Gender.male,
        roles=[Role.user]
         ),
    User(
        id=UUID("3f4f106b-1758-4e1f-a5ac-205ad4a177bf"),
        first_name="Natalia",
        last_name="Landofdawn",
        gender=Gender.female,
        roles=[Role.user,Role.admin]
         )
]

@app.get('/')
async def root():
    return {"hello":"world"}

@app.get('/api/v1/users')
async def fetch_users():
    return db

@app.post("/api/v1/users'")
async def register_user(user: User):
    db.append(user)
    return {"id":user.id}`

models.py
`from typing import List
from fastapi import FastAPI
from uuid import UUID,uuid4
from models import User, Gender, Role

app = FastAPI()

db: List[User] = [
    User(
        id=uuid4(),
        first_name="John",
        last_name="Doe",
        gender=Gender.male,
        roles=[Role.user]
         ),
    User(
        id=UUID("3f4f106b-1758-4e1f-a5ac-205ad4a177bf"),
        first_name="Natalia",
        last_name="Landofdawn",
        gender=Gender.female,
        roles=[Role.user,Role.admin]
         )
]

@app.get('/')
async def root():
    return {"hello":"world"}

@app.get('/api/v1/users')
async def fetch_users():
    return db

@app.post("/api/v1/users'")
async def register_user(user: User):
    db.append(user)
    return {"id":user.id}`

### Operating System

Windows

### Operating System Details

Windows 10

### FastAPI Version

fastapi 0.89.0

### Python Version

Python 3.11.0

**Comment by visuxls** [comment, marked as answer]
I think it's because your route here: `@app.post("/api/v1/users'")` has an extra apostrophe. Edit it to `@app.post("/api/v1/users")` and should work.

**Comment by muhiqsimui** [comment]
> I think it's because your route here: `@app.post("/api/v1/users'")` has an extra apostrophe. Edit it to `@app.post("/api/v1/users")` and should work.

Thank you sir it's working now
